packages/database/src/core/WriteTree.ts

```diff
@@ -504,29 +493,18 @@ export class WriteTree {
     }
   }
 
-  /**
-   * @param {!WriteRecord} writeRecord
-   * @param {!Path} path
-   * @return {boolean}
-   * @private
-   */
   private recordContainsPath_(writeRecord: WriteRecord, path: Path): boolean {
     if (writeRecord.snap) {
       return writeRecord.path.contains(path);
     } else {
-      // findKey can return undefined, so use !! to coerce to boolean
-      return !!findKey(writeRecord.children, function(
-        childSnap: Node,
-        childName: string
-      ) {
-        return writeRecord.path.child(childName).contains(path);
-      });
+      return Object.keys(writeRecord.children).some((childName: string) =>
```

Comment:
This creates a copy now. 

It's also not supported by IE11 and needs another Polyfill: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
Done. I'm sure that at least 99% of people who use this just load all polyfills and not pick them one by one though.